docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+
+```bash
+~/code/uvisor-example/.build/K64F_SECURE/GCC_ARM/uvisor-example.bin
+```
+
+Re-flash the device and press the reset button. The device LED should be blinking as in the previous case.
+
+## Add a secure box
+[Go to top](#overview)
+
+Now that uVisor is enabled, we can finally add a *secure box*.
+
+A secure box is a special compartment that is granted exclusive access to peripherals, memories and interrupts. Private resources are only accessible when the *context* of the secure box is active. The uVisor is the only one that can enable a secure box context, for example upon thread switching or interrupt handling.
+
+Code that belongs to a box is not obfuscated by uVisor, so it is still readable and executable from outside of the box. In addition, declaring an object in the same file that configures a secure box does not protect that object automatically.
+
+Instead, we provide specific APIs to instruct the uVisor to protect a private resource. Here we will show how to use these APIs in the `uvisor-example` app.
+
+### Configure the secure box
+
+For this example, we want to create a secure box called `private_timer`. The `private_timer` box has exclusive access to the PIT timer and to the GPIO PORT C on the NXP FRDM-K64F board. These peripherals cannot be accessed from outside the secure box.
+
+Each secure box must have at least one thread, which we call the box main thread. In our `private_timer` box we will only use this thread throughout the whole program. The thread will constantly save the current timer value in a private buffer. In addition, we want to print the content of the buffered timer values whenever we press the button on the board.
+
+We want the box to have exclusive access to the following resources:
+
+* The timer and push-button peripherals (peripheral ACLs). Nobody else should be able to read the timer values.
+* The push-button interrupt (IRQ ACL). We want the button IRQ to be re-routed to our box-specific ISR.
+* The buffer that holds the timer samples (dynamic memory ACL).
+* The static memory that holds information about the timer buffer (static memory ACL).
+
+Create a new source file, `~/code/uvisor-example/source/secure_box.cpp`. We will configure the secure box inside this file. The secure box name for this example is `private_timer`.
+
+```C
+/* ~/code/uvisor-example/source/secure_box.cpp */
+
+#include "mbed.h"
+#include "rtos.h"
+#include "uvisor-lib/uvisor-lib.h"
+
+/* Private static memory for the secure box */
+typedef struct {
+    uint32_t * buffer;
+    int index;
+} PrivateTimerStaticMemory;
+
+/* ACLs list for the secure box: Timer (PIT). */
+static const UvisorBoxAclItem g_private_timer_acls[] = {
+    {PIT,   sizeof(*PIT),   UVISOR_TACLDEF_PERIPH}
+    {PORTC, sizeof(*PORTC), UVISOR_TACLDEF_PERIPH},
+};
+
+static void private_timer_main_thread(const void *);
+
+/* Secure box configuration */
+UVISOR_BOX_NAMESPACE(NULL);                   /* Box namespace. It can be NULL for this example. */
```

Comment:
> `It can be NULL for this example.`

It can be NULL for any other example, or any other application, too.

`We won't specify a namespace for this example.`